### ISSUE TITLE
Data Feed for Twitch Chat

### ISSUE THREAD

**Issue by JackTheFoxOtter** [Yellow-Dog-Man/Resonite-Issues#2381]
### Is your feature request related to a problem? Please describe.

Currently, creating custom Twitch chat dialogs is somewhat limited. You have two options, using the TwitchChatDialog, or using the TwitchChatMessageEvent ProtoFlux node. Both have different limitations:
- The TwitchChatDialog creates UI elements from code, so it's not possible to customize templates for the individual chat messages
- The TwitchChatMessageEvent ProtoFlux node can be used to instantiate custom chat message templates, but it's missing the handling of custom emotes the TwitchChatDialog component provides through a managed DynamicSpriteFont

### Describe the solution you'd like

I'd like to see the TwitchChatDialog to be reworked into a data feed. That would allow users to create custom chat message templates, while maintaining the functionality of the managed DynamicSpriteFont to support custom chat emotes. Ideally the "raw" message content and the content using custom sprite RTF could be provided independantly.

### Describe alternatives you've considered

Somehow using the TwitchChatMessageEvent and TwitchChatDialog together to take the dynamic sprite font from the TwitchChatDialog but build a custom UI through the TwitchChatMessageEvent, which could work if the TwitchChatMessageEvent was extended to also export a version of the message containing RTF for custom emotes, but I don't think this would be a very elegant solution. DataFeeds would make it much easier.

### Additional Context

It might make sense to consider the possibility of other chat providers being supported in the future (YouTube, Kick etc.) and keeping the underlying functionality somewhat generic. I've also seen a trend of "multistreams" becoming more common, so the ability to merge multiple chat providers into a single feed of messages could potentially be useful as well. I'm not requesting that right now, but I wanted to bring it up for consideration at least.

**Comment by shiftyscales** [comment]
This is effectively a sub-issue of #2342:

That issue specifically mentions:
> - Splitting out Twitch to its own modular UI pieces

If you have specific points of feedback for that issue, I'd recommend contributing on that issue instead of this one separately to that, @JackTheFoxOtter.

**Comment by JackTheFoxOtter** [comment]
This is talking about the Camera control panel. Not the Twitch chat integration.

**Comment by shiftyscales** [comment]
The Twitch chat is one of the parts of the camera control panel per the point I highlighted above, @JackTheFoxOtter.

**Comment by JackTheFoxOtter** [comment]
Yes, but the issue you linked isn't talking about using Data Feeds to be used in the Twitch chat integration, it's focused on restructuring the streaming camera UI. My request is not related to the streaming camera UI, but the Twitch chat integration in general. Which is also used within by the streaming camera UI, but I'd rather keep that issue focused on restructuring the UI, and this one about the way the Twitch chat integration works, which is not affected at all by the structure of the streaming camera UI.

**Comment by shiftyscales** [comment]
"Splitting out Twitch to its own modular UI pieces" would entail redesigning the Twitch chat to make use of modular pieces / systems. As noted- if you had specific questions or requests about that work- it'd be best to post them on the issue linked.

**Comment by JackTheFoxOtter** [comment]
So you'd rather we have one big issue that contains a bunch of loosely related requests instead of individual small issues focused on isolated requests? Because at this point I feel like if I report a large issue, I get told to split it up, and if I report a small issue, I get told to merge it into a loosely related larger one. I'm not sure what to report at all anymore.

Considering the Twitch chat has nothing to do with the streaming camera UI except the fact that it happens to be used inside of it, I don't want to just add to that issue, and instead make an isolated one, which is what the reporting guidelines suggest. I'd like to see the Twitch chat be reworked independantly of the streaming camera UI.

**Comment by JackTheFoxOtter** [comment]
Can this issue please get reevaluated? I still don't believe it was closed rightfully.

**Comment by Frooxius** [comment]
We can look at this independently of the streaming camera UI rework. While we'd probably do this as part of the rework, having a data feed for Twitch Chat is a standalone enough item that can be done ahead of time - especially since this will allow building custom Twitch chat UI's, before the whole of streaming is reworked.

It's similar what we did with the worlds data feed - it's a building block that will be used for reworking the Worlds UI, but it's also useful item on its own to build other UI's and systems, so we implemented part of it already, while the whole Worlds UI rework is still waiting.
